components/tabs/__tests__/index.test.tsx

```diff
@@ -50,73 +42,169 @@ describe('Tabs', () => {
+        </Tabs>
+      )
+    }
+    const wrapper = mount(<App />)
+    wrapper.unmount()
   })
 
-  it('should be activate the specified tab', async () => {
-    const wrapper = mount(
+  it('can be controlled', () => {
+    mount(
       <Tabs value="1">
         <Tabs.Item label="label1" value="1">
-          test-1
+          1
         </Tabs.Item>
-        <Tabs.Item label="label2" value="2">
-          test-2
+        <Tabs.Item label="label2" value="2" disabled>
+          2
         </Tabs.Item>
       </Tabs>,
     )
-    let active = wrapper.find('header').find('.active')
-    expect(active.text()).toContain('label1')
-
-    wrapper.setProps({ value: '2' })
-    await updateWrapper(wrapper, 350)
-    active = wrapper.find('header').find('.active')
-    expect(active.text()).toContain('label2')
   })
 
-  it('should warning when label duplicated', () => {
-    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
-    mount(
-      <Tabs>
+  it('should ignore events when tab disabled', () => {
+    const changeHandler = jest.fn()
+    const wrapper = mount(
+      <Tabs initialValue="1" onChange={changeHandler}>
         <Tabs.Item label="label1" value="1">
-          test-1
+          1
         </Tabs.Item>
-        <Tabs.Item label="label2" value="1">
-          test-2
+        <Tabs.Item label="label2" value="2" disabled>
+          2
         </Tabs.Item>
       </Tabs>,
     )
-    expect(errorSpy).toHaveBeenCalled()
-    errorSpy.mockRestore()
+
+    wrapper.find('header').find('.tab').at(1).simulate('click', nativeEvent)
+    expect(changeHandler).not.toHaveBeenCalled()
+  })
+
+  it('should return default nav', () => {
+    nav({})
   })
 
-  it('should use label as key when value is missing', async () => {
-    const errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
+  it('should ingnore click in uncontroled component', () => {
     const wrapper = mount(
-      <Tabs>
-        <Tabs.Item label="label1">test-1</Tabs.Item>
-        <Tabs.Item label="label2">test-2</Tabs.Item>
+      <Tabs initialValue="1" value="2" showDivider={true} varient="solid">
+        <Tabs.Item label="label1" value="1">
+          1
+        </Tabs.Item>
+        <Tabs.Item label="label2" value="2" disabled>
+          2
+        </Tabs.Item>
       </Tabs>,
     )
-    expect(errorSpy).not.toHaveBeenCalled()
+    wrapper.find('header').find('.tab').at(0).simulate('click', nativeEvent)
+  })
+  it('should be able to update tab-item props', () => {
+    function App() {
+      const [label1, setLabel1] = useState('label1')
+      useEffect(() => {
+        setLabel1('label11')
+      }, [])
+      return (
+        <Tabs initialValue="1" value="2" showDivider={true}>
+          <Tabs.Item label={label1} value="1">
+            1
+          </Tabs.Item>
+          <Tabs.Item label="label2" value="2" disabled>
+            2
+          </Tabs.Item>
+        </Tabs>
+      )
+    }
+    mount(<App />)
```

Comment:
There's no expects or asserts here

Reply:
Currently, just make sure the code is covered, not useful test though